### PR TITLE
Refactor project code format and add CI configs

### PR DESCRIPTION
**Pull request by Xanonymous-GitHub**
## Description
- The original code of the project has no fixed format and is messy, so I added `eslint` and `prettier` for unified formatting.
  So you can check & fix code format by using `pnpm lint`, `pnpm lint:fix` and `pnpm format` after this PR.

- In addition, I also added the `dockerfile` that will be used for deployment, as well as the CI configuration file.

- And move asset files into `src/assets` folder. (json, css, images)

Also, since the CI config file has not been merged into the main branch, the CI check results cannot be seen here.
If it runs failed later, I will open another PR to fix it.

## How to verify?
Since almost all the files are reformatted, it is recommended not to view them one by one, as you may not be able to read them all.

- run dev server to see if it looks OK (go to every pages, click every buttons, etc).
- run production build and `pnpm preview` to see if the production builds normally.

### CODE REVIEW

**Comment by lohsuan** [comment]
It looks nice to me.

File: src/components/Welcome.jsx [hunk 1]
```diff
@@ -1,128 +1,132 @@
...
-        className="my-2 w-full rounded-sm p-2 outline-none bg-transparent text-white border-none text-sm white-glassmorphism"
-    />
-);
+  <input
+    placeholder={placeholder}
+    type={type}
+    step="0.0001"
+    value={value}
+    onChange={(e) => handleChange(e, name)}
+    className="my-2 w-full rounded-sm p-2 outline-none bg-transparent text-white border-none text-sm white-glassmorphism"
```
Comment (tzulin) [inline src/components/Welcome.jsx]: It runs well on dev server and can build normally for me, all functions we had before could work, including connecting the wallet, donateForFood, getAllPuppies and getAllTransactions.

Every pages look well. But there's some difference in the Welcome page. I think the input part looks prettier without the "white-glassmorphism" style. We have added this style before too, but some how it didn't work on my page before, and I think it's prettier without it. I suggest to remove it but it's also fine to keep it.

```suggestion
    className="my-2 w-full rounded-sm p-2 outline-none bg-transparent text-white border-none text-sm"
```

**Review by tzulin — APPROVED** [review]
Though I haven't try the `dockerfile`, it looks fine to me. I think maybe we could try it when we're about to deploy.

These changes looks good to me. Thank you for the effort.

File: src/components/Welcome.jsx (hunk 1 above)
Comment (Xanonymous-GitHub) [reply src/components/Welcome.jsx]: Nice advice
addressed in [3c075b9](https://github.com/lohsuan/puppy_sponsor/pull/5/commits/3c075b93627ffbfe611237617f100ad6705e21c2).

**Pull request merged**